File: Objective-C/Neocom/Neocom/NCDatabaseTypeInfoHeaderViewController.m
//
//  NCDatabaseTypeInfoHeaderViewController.m
//  Neocom
//
//  Created by Artem Shimanski on 25.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "NCDatabaseTypeInfoHeaderViewController.h"
#import "NCDatabase.h"
#import "UIColor+CS.h"
#import "NSAttributedString+NC.h"
@import CoreText;


@interface NCDatabaseTypeInfoHeaderViewController ()

@end

@implementation NCDatabaseTypeInfoHeaderViewController

- (void)viewDidLoad {
    [super viewDidLoad];
	self.titleLabel.text = self.type.typeName ?: NSLocalizedString(@"Unknown", nil);
	/*NSString* typeName = self.type.typeName ?: NSLocalizedString(@"Unknown", nil);
	NSMutableAttributedString* title = [[NSMutableAttributedString alloc] initWithString:[NSString stringWithFormat:@"%@ %d", typeName, self.type.typeID]];
	NSRange typeIDRange = NSMakeRange(typeName.length + 1, title.length - typeName.length - 1);
	[title addAttributes:@{NSFontAttributeName: [self.titleLabel.font fontWithSize:self.titleLabel.font.pointSize * 0.4],
						  // (__bridge NSString*) (kCTSuperscriptAttributeName): @(-1),
						   NSForegroundColorAttributeName: [UIColor lightTextColor]}
				   range:typeIDRange];
	

	
	
	self.titleLabel.attributedText = title;*/
	self.textView.attributedText = [self.type.typeDescription.text attributedStringWithDefaultFont:self.textView.font textColor:self.textView.textColor];
	self.imageView.image = (id) self.type.icon.image.image ?: NCDBEveIcon.defaultTypeIcon.image.image;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Neocom/Neocom/NCDatabaseTypeInfoHeaderViewController.swift
//
//  NCDatabaseTypeInfoHeaderViewController.swift
//  Neocom
//
//  Created by Artem Shimanski on 08.12.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

class NCDatabaseTypeInfoHeaderViewController: UIViewController {
	@IBOutlet weak var imageView: UIImageView!
	@IBOutlet weak var titleLabel: UILabel!
	@IBOutlet weak var subtitleLabel: UILabel!
	@IBOutlet weak var textView: UITextView!
	
	var type: NCDBInvType?
	var image: UIImage?
	
	override func viewDidLoad() {
		super.viewDidLoad()
		self.view.autoresizingMask = [.flexibleWidth, .flexibleBottomMargin]
		titleLabel.text = type?.typeName
		
		var groups = [String]()
		var marketGroup = type?.marketGroup
		while let marketGroupName = marketGroup?.marketGroupName {
			groups.insert(marketGroupName, at: 0)
			marketGroup = marketGroup?.parentGroup
		}

		if !groups.isEmpty {
			subtitleLabel.text = groups.joined(separator: " / ")
		}
		else if let group = type?.group?.groupName, let category = type?.group?.category?.categoryName {
			subtitleLabel.text = "\(category) / \(group)"
		}
		else {
			subtitleLabel.text = nil
		}
		textView.attributedText = type?.typeDescription?.text?.withFont(textView.font!, textColor: textView.textColor!)
		textView.linkTextAttributes = [NSForegroundColorAttributeName: UIColor.caption]
		imageView.image = image ?? type?.icon?.image?.image ?? NCDBEveIcon.defaultType.image?.image
	}
	
}
